control-plane-operator/controllers/hostedcontrolplane/ingressoperator/ingressoperator.go

```diff
@@ -0,0 +1,115 @@
+package ingressoperator
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/kas"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/support/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilpointer "k8s.io/utils/pointer"
+)
+
+type Params struct {
+	IngressOperatorImage    string
+	HAProxyRouterImage      string
+	KubeRBACProxyImage      string
+	ReleaseVersion          string
+	TokenMinterImage        string
+	AvailabilityProberImage string
+}
+
+func NewParams(version string, images map[string]string) Params {
+	return Params{
+		IngressOperatorImage:    images["cluster-ingress-operator"],
+		HAProxyRouterImage:      images["haproxy-router"],
+		ReleaseVersion:          version,
+		TokenMinterImage:        images["token-minter"],
+		AvailabilityProberImage: images[util.AvailabilityProberImageName],
+	}
+}
+
+func ReconcileDeployment(dep *appsv1.Deployment, params Params, apiPort *int32) {
```

Comment:
Since we're going to manage this like other control plane components, should we apply the usual set of generic configuration bits to it like we do other components, e.g.

```
DeploymentConfig.SetColocation()
DeploymentConfig.SetRestartAnnotation()
DeploymentConfig.SetControlPlaneIsolation()
DeploymentConfig.SetMultizoneSpread()
```
And replicas (and MultiZoneSpread) based on ControllerAvailabilityPolicy, etc?

Reply:
@alvaroaleman @csrwng looks like the latest commit addresses most of these, but there's no HA support... will we want HA for this one or not?

Reply:
I copied this from the CVO which also always runs with one replica. I don't think there is really much value in keeping a stand-by replica running?

Reply:
I don't know we need standbys either, just calling it out for posterity in case someone else has a reason. thanks!